### DISCUSSION TITLE
No Channels Found/Channel Scan Error

### DISCUSSION THREAD

**Discussion by OliRavioli** [tgorgdotcom/locast2plex#252] [Q&A] [answered]
I have installed Locast2Plex on a Win10 box which is running VirtualBox.  The Locast2Plex service loads successfully on the Win10 box, and when I go into the "Live TV & DVR" setting on Plex the system recognizes Locast2Plex as a network device.  The problem I have is when I attempt to scan for channels, Plex immediately gives this error:  "No channels found. Make sure your device has a signal from your antenna and try scanning again."  

The Locast2Plex service is running and receives the request from Plex to scan channels, as I see it respond from it's service window on Win10.  When I attempt to scan for channels, the Locast2Plex service reports that it is:
- "Checking FCC facilitiies database for updates"
- "Cached facilities database is current"
- "Found 28 stations for DMA 511"

I am using the local IP address for the WIN10 system in the Locast2Plex config.ini.  I tried using the VirtualBox Host-Only network IP address as a test, but Plex couldn't see the Locast2Plex service.  

Any suggestions on what I can do to resolve this issue would be greatly appreciated!

**Comment by OliRavioli** [comment]
I have since removed VirtualBox from the system so that I'm running Locast2Plex directly from Win10, but I continue to get the same error when attempting to scan for channels from Plex.

**Comment by pvtsnuffy** [comment]
@OliRavioli 
I had this exact same issue when using Docker on WIN10. I ended up blowing out Docker and am now running the python script directly on WIN10 and the issue completely went away. A few of my actions that seemed to have helped resolve my issue:

1. Install Python via Microsoft Store by typing "python" in a command prompt and uninstall all other versions
2. Download the latest locast2plex release files on GitHub and extract to C:\locast2plex
3. I am using locast2plex on the same machine as Plex so I put the command "plex_accessible_ip = 127.0.0.1" in the config.ini file
4. The only other things I changed were the username and password fields

I've been running it for 3 days now and it has been absolutely rock solid; even with increasing the tuners to 4 to record 4 shows at once while concurrently streaming 3 HD videos. No issues.

**Reply by OliRavioli** [reply]
No joy.  I tried all of your suggestions, but I still can't get plex to successfully scan for channels.  The only difference that I see between our setups is that I'm running Plex on a separate machine.  

When I request a channel scan from Plex I see the Locast2Plex service immediately react, but it looks like the response doesn't make it back to the Plex server.  I've tried adding port forwarding on my router, as well as on the Win10 firewall but the problem persists.  I even went so far as to completely disable the Win 10 firewall but it still didn't work.

Frustrating... I much rather get this working instead of buying a tuner to use with Plex.

**Reply by OliRavioli** [reply]
![image](https://user-images.githubusercontent.com/48881870/115964523-e7077a80-a4f2-11eb-8c50-aa2b096369b9.png)

**Reply by OliRavioli** [reply]
This is what my Locast2Plex service displays when I attempt to scan channels.

**Reply by pvtsnuffy** [reply]
@OliRavioli Your IP should be showing 127.0.0.1 not 192.168.1.4 if both plex and locast2plex are running on the same machine. Here is what my config.ini and command prompt look like with my username and password removed.

![Capture2](https://user-images.githubusercontent.com/72513367/115965567-e3c2bd80-a4f7-11eb-9f2f-5a4919716c37.PNG)
![Capture](https://user-images.githubusercontent.com/72513367/115965572-e7eedb00-a4f7-11eb-844f-9b2f9f6ea81c.PNG)

**Reply by OliRavioli** [reply]
Thanks @pvtsnuffy, but I'm running Plex on a separate Unix box.  Locast2Plex is running on a physically separate mini-PC with Win10.  

I see you have tuner_count = 4 in your config.  Do you know what that value equates to?  I see my service defaults to a tuner count of 3 when it starts.

**Reply by OliRavioli** [reply]
I just compared our log files.  It looks like your system moves to "Starting Stations thread" after finding 22 stations for your DMA, and then starts an SSDP server.  My system never gets this far.  That helps explain why my Plex system is unable to scan for channels.  Now, what would cause my system to get hung up at this point...

**Comment by CTJohnK** [comment]
I believe the 'plex_accessible_ip' should be the IP of the system running locast2plex.

What are the IP addresses of your Unix box and that of Locast2Plex?

**Reply by CTJohnK** [reply]
I forgot to mention, that setting tells Plex how many concurrent sessions it can pull from Locast2Plex.  Four concurrent sessions is the maximum Locast allows.   So, if you have two people watching Locast directly and Plex tries to record 3 shows at the same time, one of those shows will not be recorded.

tuner_count = 4

**Reply by OliRavioli** [reply]
Thanks @CTJohnK.  The Win10 system hosting Locast2Plex has the IP address 192.168.1.240.  The Plex server is a Linux box with an IP of 192.168.1.4.

**Comment by CTJohnK** [comment]
Then use 192.168.1.240On Apr 24, 2021 5:23 PM, OliRavioli ***@***.***> wrote:
Thanks @CTJohnK.  The Win10 system hosting Locast2Plex has the IP address 192.168.1.240.  The Plex server is a Linux box with an IP of 192.168.1.4.

**Reply by OliRavioli** [reply]
Thanks - I was using the correct IP.

**Comment by OliRavioli** [comment, marked as answer]
I finally got the channel scan to work.  I don't know exactly what the problem was, but after trying multiple times to get the Locast2Plex service to run correctly from the Win10 command line, I was able to get it to work using Docker.  Overall steps that I went thru included:
- Loading Locast2Plex from Win10 command line
- Running Live TV & DVR configuration within Plex.  Locast2Plex appeared as a tuner, but attempts to scan channels failed
- Loading Locast2plex from Docker
- Rerunning Live TV & DVR configuration within Plex.  Locast2Plex still appeared as a tuner, and attempts to scan channels still failed
- KEY ITEM:  I manually added the IP address for Locast2Plex on the Live TV & DVR configuration page.  A second Locast2Plex tuner box appeared (so now my screen showed 2 separate Locast2Plex tuners).  I clicked on the new tuner box and was able to successfully scan for channels

I don't know why the Docker container works when the Win10 command line doesn't, but at least it is working!
